File: Patrones de Diseño/Creacionales/Fábrica abstracta/src/Ejemplo - Interfaz de usuario multiplataforma/Aplicacion.java
import Componentes.Botones.Boton;

class Aplicacion {
	private FabricaGUI fabrica;
	private Boton boton;

	Aplicacion(FabricaGUI fabrica) {
		this.fabrica = fabrica;
	}

	void crearInterfazDeUsuario() {
		this.boton = this.fabrica.crearBoton();
	}

	void pintar() {
		this.boton.pintar();
	}
}

File: Patrones de Diseño/Creacionales/Fábrica abstracta/src/Ejemplo - Interfaz de usuario multiplataforma/Aplicacion.php
<?php

use Componentes\Boton;
use GUI\FabricaGUI;

class Aplicacion {
	private Boton $boton;

	function __construct(private FabricaGUI $fabrica) {}

	function crearUI() {
		$this->boton = $this->fabrica->crearBoton();
	}

	function pintar() {
		$this->boton->pintar();
	}
}
